pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/inv/json/BaseJsonIndexCreator.java

```diff
@@ -87,7 +89,15 @@
   @Override
   public void add(String jsonString)
       throws IOException {
-    addFlattenedRecords(JsonUtils.flatten(JsonUtils.stringToJsonNode(jsonString)));
+    JsonNode jsonNode = JsonUtils.stringToJsonNode(jsonString);
```

Comment:
We don't really need another validation here. It is already validated during the type conversion (PinotDataType)